File: January 2023/31 January 2023/Minimum times A has to be repeated such that B is a substring of it.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
//User function Template for C++

class Solution {
  public:
    int minRepeats(string A, string B) {
        string rep=A;
        int cnt=1;
        while(A.size()<B.size()){
            A+=rep;
            cnt++;
        }
        if(A.find(B)!=-1){
            return cnt;
        }
        A+=rep;
        cnt++;
        if(A.find(B)!=-1){
            return cnt;
        }
        return -1;
    

    }
};

//{ Driver Code Starts.

int main() {
    int t;
    scanf("%d ",&t);
    while (t--) {
        string A,B;
        getline(cin,A);
        getline(cin,B);

        Solution ob;
        cout << ob.minRepeats(A,B) << endl;
    }
    return 0;
}
// } Driver Code Ends

File: January 2023/31 January 2023/Minimum times A has to be repeated such that B is a substring of it.py
class Solution:
    def issubstring(self, str2, rep):
        if str2 in rep:
            return True
        return False
    
    def minRepeats(self, A, B):
        ans = 1
        S = A
        
        while len(S)<len(B):
            S += A 
            ans += 1 
            
        if self.issubstring(B,S):
            return ans 
        elif self.issubstring(B,S+A):
            return ans+1 
        else:
            return -1

#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        A=input()
        B=input()
        
        ob = Solution()
        print(ob.minRepeats(A,B))
# } Driver Code Ends
